swarm/src/handler.rs

```diff
@@ -224,6 +277,76 @@ pub trait ConnectionHandler: Send + 'static {
     {
         ConnectionHandlerSelect::new(self, other)
     }
+
+    /// informs the handler about an event coming from the outside in the handler.
+    fn on_behaviour_event(&mut self, _event: Self::InEvent) {}
+
+    fn on_connection_event(
+        &mut self,
+        _event: ConnectionEvent<
+            Self::InboundProtocol,
+            Self::OutboundProtocol,
+            Self::InboundOpenInfo,
+            Self::OutboundOpenInfo,
+        >,
+    ) {
+    }
+}
+
+/// Enumeration with the list of the possible stream events
+/// to pass to [`on_event`](ConnectionHandler::on_event).
+pub enum ConnectionEvent<'a, IP: InboundUpgradeSend, OP: OutboundUpgradeSend, IOI, OOI> {
+    /// Informs the handler about the output of a successful upgrade on a new inbound substream.
+    FullyNegotiatedInbound(FullyNegotiatedInbound<IP, IOI>),
+    /// Informs the handler about successful upgrade on a new outbound stream.
```

Comment:
```suggestion
    /// Informs the handler about the output of a successful upgrade on a new outbound stream.
```

To be consistent with the above.

Reply:
yeah makes sense, thanks Max!